pkg/commands/git_commands/stash.go

```diff
@@ -109,3 +121,26 @@ func (self *StashCommands) SaveStagedChanges(message string) error {
 
 	return nil
 }
+
+func (self *StashCommands) Rename(index int, message string) error {
+	output, err := self.Drop(index)
+	if err != nil {
+		return err
+	}
+
+	// `output` is in the following format:
+	// Dropped refs/stash@{0} (f0d0f20f2f61ffd6d6bfe0752deffa38845a3edd)
+	stashShaPattern := regexp.MustCompile(`\(([0-9a-f]+)\)`)
```

Comment:
Just tested this out locally, and it works, but I think we should change the approach a bit. You can use `git rev-parse stash@{<index>}` to obtain the hash, and I think we ought to do that for a couple of reasons:
1) `git rev-parse`'s output is never going to change, but I can imagine the output of `git stash drop` changing. If `git stash drop` does change its output, we'll have just lost the user's stash which is going to make them very mad.
2) it spares us the complexity of regexes

So I propose we run three commands:
1) `git rev-parse stash@{<index>}` to get the hash
2) `git stash drop stash@{<index>}` to drop it
3) `git stash store <hash>` to store the renamed stash entry

Reply:
changed that way:)